### PR TITLE
Reduce coupling for exception handling in Storage and Find Function

### LINKED ISSUE #170: [PE-D][Tester C] User cannot see pairing
- User don't know about the pairing unless they go to the data

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Than-Duc-Huy/ped#5

### LINKED ISSUE #187: Exceptions used in Storage features contains too much coupling.

### PR DESCRIPTION
**Pull request by wilsonngja**
Fix #170 and #187 

While improving the exceptions, #170 was also fixed as it was related and can be rectified with minor change to the code.

### CODE REVIEW

File: src/main/java/seedu/duke/Storage.java [hunk 1]
```diff
@@ -75,21 +89,28 @@ public class Storage {
     private static final String PROPERTY_ADDRESS_FLAG = " a/";
     private static final String PROPERTY_RENTAL_FLAG = " p/";
     private static final String PROPERTY_TYPE_FLAG = " t/";
+    private static final String NEXT_LINE = System.lineSeparator();
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/Storage.java:92]: Good job using System.lineSeperator() to make newlines system-independent

File: src/main/java/seedu/duke/Storage.java [hunk 2]
```diff
@@ -252,35 +283,29 @@ public void loadClient(File clientFile) {
...
 
-                    String unitTypeLabel = getUnitTypeLabel(unitTypeString);
+        Scanner scanner = new Scanner(propertyFile);
+        while (scanner.hasNext()) {
+            String[] propertyParameters = scanner.nextLine().split("\\s\\|\\s");
+            if (propertyParameters.length == 4) {
+                String landlordName = propertyParameters[0];
+                String address = propertyParameters[1];
+                String price = propertyParameters[2];
+                String unitTypeString = propertyParameters[3].trim();
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/Storage.java]: Would be good to avoid using magic literals here

File: src/main/java/seedu/duke/Storage.java [hunk 3]
```diff
@@ -290,11 +315,12 @@ public void loadProperty(File propertyFile) {
      *
      * @param pairFile The file that contains the pairing file.
      */
-    public void loadPair(File pairFile) {
+    public void loadPair(File pairFile) throws FileNotFoundException {
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/Storage.java:332]: Good job throwing an exception from this method to capture file loading irregularities in the method calling it. Helps with SLAP

File: src/main/java/seedu/duke/Storage.java [hunk 4]
```diff
@@ -499,98 +494,293 @@ public void updatePair() {
...
                         }
-
+                    } catch (DukeException e) {
+                        UI.showExceptionMessage(e);
                     }
+                } else {
+                    skipPairingEntries();
                 }
+            } else {
+                skipPairingEntries();
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/Storage.java:529]: There seems to be some arrowhead code here

File: src/main/java/seedu/duke/exception/storagexception/PairFileNotFoundException.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package seedu.duke.exception.storagexception;
```
Comment (ngdeqi) [inline src/main/java/seedu/duke/exception/storagexception/PairFileNotFoundException.java:1]: Might want to correct the typo in the package name

**Comment by wilsonngja** [comment]
This pull request has also expanded to improve to reduce coupling in Find function, thus fixing #188

File: src/main/java/seedu/duke/Storage.java (hunk 4 above)
Comment (wilsonngja) [reply src/main/java/seedu/duke/Storage.java:529]: Arrowhead code has been fixed in the latest commit. Thanks for identifying :)

**Review by FeliciaBeatrice — APPROVED** [review]
looks good! LGTM

**Comment by wilsonngja** [comment]
Fixed #189 as well.

**Pull request merged**